llama-index-core/llama_index/core/llms/mock.py

```diff
@@ -334,9 +342,21 @@ async def _gen() -> AsyncGenerator[ChatResponse, None]:
         return _gen()
 
     async def astream_chat_with_tools(
-        self, tools: List[Any], chat_history: List[ChatMessage], **kwargs: Any
+        self,
+        tools: Sequence["BaseTool"],
+        user_msg: Optional[Union[str, ChatMessage]] = None,
+        chat_history: Optional[List[ChatMessage]] = None,
+        verbose: bool = False,
+        allow_parallel_tool_calls: bool = False,
+        tool_required: bool = False,
+        **kwargs: Any,
     ) -> ChatResponseAsyncGen:
-        content = self.blocks_to_content_callback(chat_history[-1].blocks, self.tool_calls)
+        contents = chat_history or [
+            user_msg
+            if isinstance(user_msg, ChatMessage)
+            else ChatMessage(content=user_msg or "", role="user")
+        ]
```

Comment:
@AstraBert This is what I was seeing! Can we give preference to the user_msg over the chat_history? That is, `blocks_to_content_callback` should run on `user_msg` if it exists, otherwise fallback to the last chat_history.